src/python/bot/fuzzers/libfuzzer.py

```diff
@@ -1491,6 +1491,117 @@ def cleanse_crash(self,
       return result
 
 
+class AndroidEmulatorLibFuzzerRunner(AndroidLibFuzzerRunner):
+  """Android emulator libFuzzer runner."""
+
+  # pylint: disable=super-init-not-called
+  def __init__(self, executable_path, build_directory, default_args=None):
+    """Inits the AndroidEmulatorLibFuzzerRunner.
+
+    Args:
+      executable_path: Path to the fuzzer executable.
+      build_directory: A MinijailChroot.
+      default_args: Default arguments to always pass to the fuzzer.
+    """
+    self.exec_path = executable_path
+    self.build_dir = build_directory
+    self.def_args = default_args
+    # We don't call super().__init__() here since we need to wait until
+    # the emulator is started. It's called in _start_emulator().
+
+  def _start_emulator(self):
+    self.emu_proc = android.emulator.EmulatorProcess()
+    self.emu_proc.create(self.build_dir)
+    self.emu_proc.run()
+    android.adb.run_as_root()
+    super().__init__(
+        executable_path=self.exec_path,
+        build_directory=self.build_dir,
+        default_args=self.def_args)
+
+  def _stop_emulator(self):
+    self.emu_proc.kill()
+
+  def analyze_dictionary(self,
```

Comment:
Rather than repeating all of these, might be better to do something like (untested):

```python

def wrap_emulator(func):
  @functools.wraps(func)
  def wrapper(self, *args, **kwargs):
    self._start_emulator()
    result = func(self, *args, **kwargs)
    self._stop_emulator()

  return wrapper

analyze_dictionary = wrap_emulator(AndroidLibFuzzerRunner.analyze_dictionary)
fuzz = wrap_emulator(AndroidLibFuzzerRunner.fuzz)
...

```

Reply:
Done. I inlined the _start_emulator() and _stop_emulator() functions as well, since they aren't used anywhere else.